atom/common/promise_util.h

```diff
@@ -104,9 +104,25 @@ class Promise {
     return GetInner()->Reject(GetContext(), v8::Undefined(isolate()));
   }
 
-  template <typename ReturnType, typename... ArgTypes>
+  // Please note that using Then is effectively the same as calling .then
+  // in javascript.  This means (a) it is not type safe and (b) please note
+  // it is NOT type safe.
+  // If the base::Callback you provide here is of type void(boolean) and you
+  // resolve the promise with a string, Electron will compile successfully and
+  // then that promise will be rejected as soon as you try to use it as the
+  // mate converters doing work behind the scenes will throw an error for you.
+  // This can be really hard to trace so until either
+  //   * This helper becomes typesafe (by templating the class instead of each
+  //   method)
```

Comment:
If we're already doing this anywhere, would be great to cite an example here so that devs looking to use this function in the future will have something to crib from as an alternative to using `Then()`

Reply:
We're not doing this yet, I plan at some point to follow up and do this but it's a bigger / more time consuming fix